src/SelfUpdateCommand.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Robo;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Filesystem\Filesystem as sfFilesystem;
+
+/**
+ * Update the robo.phar from the latest github release
+ *
+ * @author Alexander Menk <alex.menk@gmail.com>
+ */
+class SelfUpdateCommand extends Command
+{
+    private $command;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('self-update')
+            ->setAliases(array( 'selfupdate' ))
+            ->setDescription('Updates the robo.phar to the latest version.')
+            ->setHelp(
+                <<<EOT
+The <info>self-update</info> command checks github for newer
+versions of robo and if found, installs the latest.
+EOT
+            );
+    }
+
+    protected function getLatestReleaseFromGithub($repository)
+    {
+        $opts = [
+            'http' => [
+                'method' => 'GET',
+                'header' => [
+                    'User-Agent: PHP'
```

Comment:
GitHub requests that the user-agent be filled out for identification purposes. Perhaps this should be a bit more specific.

Reply:
done